### PR TITLE
[PAN-1808] upgrade python to 3.12 and switch to poetry

### PR DESCRIPTION
**Pull request by danut13**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

1. Upgrade Python from 3.10 to 3.12.2.
2. Replace pip with poetry for dependency management.
3. Replace setuptools with poetry for packaging.

## QA Instructions

The upstream projects (service node, validatornode etc.) should work as before.

### CODE REVIEW

**Review by juanmanuel-tirado — DISMISSED** [review]
QA section LG2M

**Review by chandraprakash — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -31,31 +31,34 @@ Please make sure that your environment meets the following requirements:
 
 #### Python Version
 
-The Pantos Client CLI requires **Python 3.10**. Ensure that you have the correct Python version installed before the installation steps. You can download the latest version of Python from the official [Python website](https://www.python.org/downloads/).
+Pantos Common requires **Python 3.12**. Ensure that you have the correct Python version installed before the installation steps. You can download the latest version of Python from the official [Python website](https://www.python.org/downloads/).
```
Comment (markuslevonyak) [inline README.md:34]: Do we really require 3.12? Or just support it? If we require 3.12, which changes make it incompatible with 3.10?

Comment (markuslevonyak) [reply README.md:34]: See https://bitpanda.slack.com/archives/C040G3XU9S6/p1714397885229099

**Review by markuslevonyak — APPROVED** [review]

**Review by markuslevonyak — APPROVED** [review]

**Review by chandraprakash — DISMISSED** [review]

**Review by markuslevonyak — APPROVED** [review]

**Review by juanmanuel-tirado — DISMISSED** [review]
LG2M

**Pull request closed without merging**